### PR TITLE
Decouple offline_update and systemd Meson features

### PR DESCRIPTION
**Pull request by arrowd**

### CODE REVIEW

**Comment by arrowd** [comment]
Bump, can we get this in?

File: meson.build [hunk 1]
```diff
@@ -37,16 +37,14 @@ if get_option('systemd')
...
-    error('Offline updates requires Systemd, use -Dsystemd=true to enable it or -Doffline_update=false to disable offline updates')
-  endif
+  add_project_arguments ('-DHAVE_SYSTEMD_SD_JOURNAL_H=1', language: 'c')
 endif
 
 elogind = []
 if get_option('elogind')
   elogind = dependency('elogind', version: '>=229.4')
   add_project_arguments ('-DHAVE_SYSTEMD_SD_LOGIN_H=1', language: 'c')
+  add_project_arguments ('-DHAVE_SYSTEMD_SD_JOURNAL_H=1', language: 'c')
```
Comment (Conan-Kudo) [inline meson.build:47]: But elogind doesn't provide sd-journal, so why would we do that?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (arrowd) [reply meson.build:47]: Oh, it doesn't? I had no idea, I will fix that.

Comment (Conan-Kudo) [reply meson.build]: Hmm, it seems it does? https://github.com/elogind/elogind/tree/main/src/libelogind/sd-journal

Comment (arrowd) [reply meson.build]: image

File: client/pk-offline-update.c [hunk 2]
```diff
@@ -32,7 +32,13 @@
 #include <packagekit-glib2/pk-offline-private.h>
 #include <stdlib.h>
 #include <unistd.h>
+#ifdef HAVE_SYSTEMD_SD_JOURNAL_H
 #include <systemd/sd-journal.h>
+#else
+#define LOG_INFO STDERR_FILENO
+#define LOG_WARNING STDERR_FILENO
+#define sd_journal_print dprintf
+#endif
```
Comment (Conan-Kudo) [inline client/pk-offline-update.c:41]: Does this actually work? This seems a bit sketchy...

Comment (arrowd) [reply client/pk-offline-update.c:41]: Yes, it prints correctly here on FreeBSD.

Comment (Conan-Kudo) [reply client/pk-offline-update.c:41]: Huh, okay...

**Review by Conan-Kudo — APPROVED** [review]

**Pull request merged**